common/src/main/java/com/wynntils/utils/mc/StyledTextUtils.java

```diff
@@ -198,4 +207,26 @@ public static StyledText unwrap(StyledText styledText) {
 
         return StyledText.fromParts(newParts);
     }
+
+    public static Pair<String, String> extractNameAndNick(StyledText styledText) {
+        for (StyledTextPart part : styledText){
+            HoverEvent hoverEvent = part.getPartStyle().getStyle().getHoverEvent();
+
+            if (hoverEvent == null || hoverEvent.getAction() != HoverEvent.Action.SHOW_TEXT) {
+                continue;
+            }
+
+            StyledText[] partTexts = StyledText.fromComponent(hoverEvent.getValue(HoverEvent.Action.SHOW_TEXT)).split("\n");
+
+            for (StyledText partText : partTexts) {
+                Matcher nicknameMatcher = partText.getMatcher(NICKNAME_PATTERN);
+
+                if (nicknameMatcher.matches()) {
+                    return new Pair<>(nicknameMatcher.group("username"), nicknameMatcher.group("nick"));
+                }
+            }
+        }
+
+        return new Pair<>(null, null);
```

Comment:
I think in this case it's generally smarter to return null, as it makes error handling code nicer.